### PR TITLE
5166 Updating new roles and removing old roles

### PR DESCRIPTION
**Pull request by leanneeliatra**
### Description
Updates to the users and roles page to make it updated with what the user sees when using OpenSearch v2.7

### Issues Resolved
[[DOC] Openserach.org Users and roles docs are out of date #5166](https://github.com/opensearch-project/documentation-website/issues/5166)

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by leanneeliatra** [comment]
@hdhalter this is ready for review please.cc @scrawfor99

File: _security/access-control/users-roles.md [hunk 1]
```diff
@@ -139,37 +139,36 @@ See [Create role mapping]({{site.url}}{{site.baseurl}}/security/access-control/a
...
+| `point_in_time_full_access`                      | Grants full permissions to all Point in Time operations.                                                                                                                                                                                                                                                                                    |
+| `security_analytics_full_access`                 | Grants full permissions to all Security Analytics functionality.                                                                                                                                                                                                                                                                            |
+| `security_analytics_read_access`                 | Grants permissions to view the various components in Security Analytics, such as detectors, alerts, and findings. It also includes permissions that allow users to search for detectors and rules. This role does not allow a user to perform actions such as modifying or deleting a detector.                                             |
```
Comment (scrawfor99) [inline _security/access-control/users-roles.md:169]: Is this line formatting different?

Comment (leanneeliatra) [reply _security/access-control/users-roles.md:169]: My editor formatted it on save but it made it easier to read on my end so I thought it was okay, also when the screen is wide enough (as wide as the widest line) the table break symbol doesn't wrap to the next line like that. 
So it's all as expected, it compiles correctly and previews correctly and is just formatted and when looking at the code, for me it reads better. I can also remove the whitespace if you prefer.

**Comment by leanneeliatra** [comment]
@Naarcha-AWS this has been reviewed by Stephen, perhaps you can take a look? or flagging for next steps please @hdhalter

**Comment by Naarcha-AWS** [comment]
@leanneeliatra: I'll have some comments for your today.

**Comment by hdhalter** [comment]
Thanks, @leanneeliatra! Can you please put the items in the table in alphabetical order? Also, can you make the TOC at the top collapsible? Here is the code:

<details closed markdown="block">
  <summary>
    Table of contents
  </summary>
  {: .text-delta }
- TOC
{:toc}
</details>

**Comment by leanneeliatra** [comment]
Thanks @hdhalter for providing the TOC code. I have made that change and also made the table listed in alphabetical order.

**Comment by hdhalter** [comment]
> Thanks @hdhalter for providing the TOC code. I have made that change and also made the table listed in alphabetical order.

Thanks, @leanneeliatra ! The table looks good. For the TOC, please remove lines 39 and 40.

**Review by Naarcha-AWS — APPROVED** [review]

**Comment by leanneeliatra** [comment]
Hi @hdhalter, I made that change thank you.

I also moved the TOC up the page, to [above the first section called out in the TOC](https://opensearch.org/docs/latest/security/access-control/users-roles/#creating-and-editing-opensearch-roles). 

To me this makes more sense than after [Editing the roles.yml file](https://opensearch.org/docs/latest/security/access-control/users-roles/#editing-the-rolesyml-file).

File: _security/access-control/users-roles.md [hunk 2]
```diff
@@ -142,34 +146,33 @@ The Security plugin includes several predefined roles that serve as useful defau
 | **Role** | **Description** |
 | :--- | :--- |
 | `alerting_ack_alerts` | Grants permissions to view and acknowledge alerts, but not to modify destinations or monitors. |
-| `alerting_full_access` | Grants full permissions to all alerting actions. |
+| `alerting_full_access` | Grants full permissions to perform all alerting actions. |
 | `alerting_read_access` | Grants permissions to view alerts, destinations, and monitors, but not to acknowledge alerts or modify destinations or monitors. |
-| `anomaly_full_access` | Grants full permissions to all anomaly detection actions. |
+| `all_access` | Grants full access to the cluster, including all cluster-wide operations, permission to write to all cluster indexes, and permission to write to all tenants. For more information about access using the REST API, see [Access control for the API]({{site.url}}{{site.baseurl}}/security/access-control/api/#access-control-for-the-api). |
```
Comment (natebower) [inline _security/access-control/users-roles.md:151]:
```suggestion
| `all_access` | Grants full access to the cluster, including all cluster-wide operations, permissions to write to all cluster indexes, and permissions to write to all tenants. For more information about access using the REST API, see [Access control for the API]({{site.url}}{{site.baseurl}}/security/access-control/api/#access-control-for-the-api). |
```

File: _security/access-control/users-roles.md [hunk 3]
```diff
@@ -142,34 +146,33 @@ The Security plugin includes several predefined roles that serve as useful defau
...
+| `anomaly_full_access` | Grants full permissions to perform all anomaly detection actions. |
 | `anomaly_read_access` | Grants permissions to view detectors, but not to create, modify, or delete detectors. |
-| `all_access` | Grants full access to the cluster, including all cluster-wide operations, permission to write to all cluster indexes, and permission to write to all tenants. For more information on access using the REST API, see [Access control for the API]({{site.url}}{{site.baseurl}}/security/access-control/api/#access-control-for-the-api). |
+| `asynchronous_search_full_access` | Grants full permissions to perform all asynchronous search actions. |
+| `asynchronous_search_read_access` | Grants permissions to view asynchronous searches, but not to submit, modify, or delete them. |
```
Comment (natebower) [inline _security/access-control/users-roles.md:155]:
```suggestion
| `asynchronous_search_read_access` | Grants permissions to view asynchronous searches but not to submit, modify, or delete them. |
```

File: _security/access-control/users-roles.md [hunk 4]
```diff
@@ -142,34 +146,33 @@ The Security plugin includes several predefined roles that serve as useful defau
...
-| `kibana_user` | Grants permissions to use OpenSearch Dashboards: cluster-wide searches, index monitoring, and write to various OpenSearch Dashboards indexes. |
-| `logstash` | Grants permissions for Logstash to interact with the cluster: cluster-wide searches, cluster monitoring, and write to the various Logstash indexes. |
-| `manage_snapshots` | Grants permissions to manage snapshot repositories, take snapshots, and restore snapshots. |
+| `cross_cluster_replication_leader_full_access`   | Grants full access to perform cross-cluster replication actions on the leader cluster. |
+| `index_management_full_access` | Grants full permissions to perform all index management actions, including Index State Management (ISM), transforms, and rollups. |
+| `index_management_read_access` | Same as `readall` but with added cluster permissions for monitoring. |
+| `ml_full_access` | Grants full permissions to perform all ML features, including starting new ML tasks and reading or deleting models. |
+| `ml_read_access` | Grants permissions to view Machine Learning features and results, but not to modify them. |
```
Comment (natebower) [inline _security/access-control/users-roles.md:161]:
```suggestion
| `ml_read_access` | Grants permissions to view ML features and results but not to modify them. |
```

File: _security/access-control/users-roles.md [hunk 5]
```diff
@@ -142,34 +146,33 @@ The Security plugin includes several predefined roles that serve as useful defau
...
-| `kibana_user` | Grants permissions to use OpenSearch Dashboards: cluster-wide searches, index monitoring, and write to various OpenSearch Dashboards indexes. |
-| `logstash` | Grants permissions for Logstash to interact with the cluster: cluster-wide searches, cluster monitoring, and write to the various Logstash indexes. |
-| `manage_snapshots` | Grants permissions to manage snapshot repositories, take snapshots, and restore snapshots. |
+| `cross_cluster_replication_leader_full_access`   | Grants full access to perform cross-cluster replication actions on the leader cluster. |
+| `index_management_full_access` | Grants full permissions to perform all index management actions, including Index State Management (ISM), transforms, and rollups. |
+| `index_management_read_access` | Same as `readall` but with added cluster permissions for monitoring. |
+| `ml_full_access` | Grants full permissions to perform all ML features, including starting new ML tasks and reading or deleting models. |
```
Comment (natebower) [inline _security/access-control/users-roles.md:160]:
```suggestion
| `ml_full_access` | Grants full permissions to perform all machine learning (ML) features, including starting new ML tasks and reading or deleting models. |
```

File: _security/access-control/users-roles.md [hunk 6]
```diff
@@ -142,34 +146,33 @@ The Security plugin includes several predefined roles that serve as useful defau
...
+| `ml_read_access` | Grants permissions to view Machine Learning features and results, but not to modify them. |
+| `notifications_full_access` | Grants full permissions to perform all Notification actions. |
```
Comment (natebower) [inline _security/access-control/users-roles.md:162]:
```suggestion
| `notifications_full_access` | Grants full permissions to perform all notification actions. |
```

File: _security/access-control/users-roles.md [hunk 7]
```diff
@@ -142,34 +146,33 @@ The Security plugin includes several predefined roles that serve as useful defau
...
+| `notifications_read_access` | Grants permissions to view notifications and their configurations, but not to modify them. |
```
Comment (natebower) [inline _security/access-control/users-roles.md:163]:
```suggestion
| `notifications_read_access` | Grants permissions to view notifications and their configurations but not to modify them. |
```

File: _security/access-control/users-roles.md [hunk 8]
```diff
@@ -142,34 +146,33 @@ The Security plugin includes several predefined roles that serve as useful defau
...
-| `snapshot_management_read_access` | Grants permissions to view policies but not to create, modify, start, stop, or delete them. |
-| `point_in_time_full_access` | Grants full permissions to all Point in Time operations. |
-| `security_analytics_full_access` | Grants full permissions to all Security Analytics functionality. |
-| `security_analytics_read_access` | Grants permissions to view the various components in Security Analytics, such as detectors, alerts, and findings. It also includes permissions that allow users to search for detectors and rules. This role does not allow a user to perform actions such as modifying or deleting a detector. |
 | `security_analytics_ack_alerts` | Grants permissions to view and acknowledge alerts. |
+| `security_analytics_full_access` | Grants full permissions to use all Security Analytics functionality. |
+| `security_analytics_read_access`  | Grants permissions to view the various components in Security Analytics, such as detectors, alerts, and findings. It also includes permissions that allow users to search for detectors and rules. This role does not allow a user to perform actions such as modifying or deleting a detector. |
```
Comment (natebower) [inline _security/access-control/users-roles.md:171]:
```suggestion
| `security_analytics_read_access`  | Grants permissions to view Security Analytics components, such as detectors, alerts, and findings. Also includes permissions that allow users to search for detectors and rules. This role does not allow a user to perform actions such as modifying or deleting a detector. |
```

File: _security/access-control/users-roles.md [hunk 9]
```diff
@@ -142,34 +146,33 @@ The Security plugin includes several predefined roles that serve as useful defau
...
+| `security_manager` | Grants permissions to manage security-related features and configurations.  |
+| `snapshot_management_full_access` | Grants full permissions to perform all snapshot management actions.  |
+| `snapshot_management_read_access`  | Grants permissions to view snapshot management actions and configurations, but not to modify them. |
```
Comment (natebower) [inline _security/access-control/users-roles.md:174]:
```suggestion
| `snapshot_management_read_access`  | Grants permissions to view snapshot management actions and configurations but not to modify them. |
```

**Review by natebower — COMMENTED** [review]
@leanneeliatra @Naarcha-AWS Please see my changes and let me know if you have any questions. Thanks!

**Comment by leanneeliatra** [comment]
Thanks for that review @natebower, you picked up some great necessary edits there, much appreciated.

The changes have all been addressed and committed now.

**Pull request merged**